sql-plugin/src/main/scala/com/nvidia/spark/rapids/basicPhysicalOperators.scala

```diff
@@ -317,8 +333,18 @@ case class GpuUnionExec(children: Seq[SparkPlan]) extends SparkPlan with GpuExec
   override def doExecute(): RDD[InternalRow] =
     throw new IllegalStateException(s"Row-based execution should not occur for $this")
 
-  override def doExecuteColumnar(): RDD[ColumnarBatch] =
-    sparkContext.union(children.map(_.executeColumnar()))
+  override def doExecuteColumnar(): RDD[ColumnarBatch] = {
+    val numOutputRows = longMetric(NUM_OUTPUT_ROWS)
+    val numOutputBatches = longMetric(NUM_OUTPUT_BATCHES)
+    val totalTime = longMetric(TOTAL_TIME)
+
+    sparkContext.union(children.map(child => {
```

Comment:
I think it would be a bit cleaner to do the metrics after the union

```
sparkContext.union(children.map(_.executeColumnar())).mapInternal { batch =>
  withResource(new NvtxWithMetrics("Union", NvtxColor.CYAN, totalTime) { time =>
    outputBatches += 1
    outputRows += batch.numRows()
    batch
  }
}
```